pmacApp/src/pmacController.cpp

```diff
@@ -4150,9 +4165,109 @@ asynStatus pmacController::readDeviceType() {
     }
   }
   debug(DEBUG_VARIABLE, functionName, "Read device CPU", cpu_);
+
+  return status;
+}
+
+asynStatus pmacController::getCpuNumCores() {
+  asynStatus status = asynSuccess;
+  char reply[PMAC_MAXBUF];
+  char cmd[PMAC_MAXBUF];
+  static const char *functionName = "getCpuNumCores";
+
+  debug(DEBUG_FLOW, functionName);
+
+  if (strcmp(cpu_.c_str(),"") == 0) {
+    status = asynError;
+    debug(DEBUG_ERROR, functionName, "Error reading card cpu");
+  }
+
+  if (status == asynSuccess) {
+    if (strcmp(cpu_.c_str(),"PowerPC,460EX") == 0) {
+      cpuNumCores_ = 1;
+    } else if (strcmp(cpu_.c_str(),"PowerPC,APM86xxx") == 0) {
+      cpuNumCores_ = 2;
+    } else if (strcmp(cpu_.c_str(),"arm,LS1021A") == 0) {
+      strcpy(cmd, "Sys.Cores");
```

Comment:
The Sys.Cores was added in V2.6 firmware, that is also the first version that supports the ARM arch.

On second thought, it might not be necessary to use Sys.Cores, since the LS1021A has a dual-core processor.
I'll try to confirm the cpu response for a quad-core processor.